# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/test/gtest/TestFileInfo.cpp
# Commit: 74d93e32c149
# Full Hash: 74d93e32c14908ff1d29cc911eca21a82f177551
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-05-06 16:28:18
# Regressor Bug: 1623278
# File Overlap Count: 1
# Description:
#   Bug 1623278 - Use SafeRefCounted in ActorsParent. r=dom-workers-and-storage-reviewers,janv
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D69351
# ==============================================================================

diff -r 6efa98635cf4 -r 74d93e32c149 dom/indexedDB/test/gtest/TestFileInfo.cpp
--- a/dom/indexedDB/test/gtest/TestFileInfo.cpp	Wed May 06 12:20:32 2020 +0000
+++ b/dom/indexedDB/test/gtest/TestFileInfo.cpp	Wed May 06 11:35:20 2020 +0000
@@ -23,13 +23,14 @@
   size_t mSyncDeleteFileCalls = 0;
 };
 
-class TestFileManager final : public FileManagerBase<TestFileManager> {
+class TestFileManager final : public FileManagerBase<TestFileManager>,
+                              public AtomicSafeRefCounted<TestFileManager> {
  public:
   using FileManagerBase<TestFileManager>::MutexType;
 
-  // FileManager functions that are used by FileInfo
+  MOZ_DECLARE_REFCOUNTED_TYPENAME(TestFileManager)
 
-  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(TestFileManager)
+  // FileManager functions that are used by FileInfo
 
   [[nodiscard]] nsresult AsyncDeleteFile(const int64_t aFileId) {
     MOZ_RELEASE_ASSERT(!mFileInfos.Contains(aFileId));
@@ -58,10 +59,8 @@
     for (const auto id : kDBOnlyFileInfoIds) {
       // Copied from within FileManager::Init.
 
-      mFileInfos.Put(
-          id, new FileInfo(FileManagerGuard{},
-                           SafeRefPtr{this, AcquireStrongRefFromRawPtr{}}, id,
-                           static_cast<nsrefcnt>(1)));
+      mFileInfos.Put(id, new FileInfo(FileManagerGuard{}, SafeRefPtrFromThis(),
+                                      id, static_cast<nsrefcnt>(1)));
 
       mLastFileId = std::max(id, mLastFileId);
     }
@@ -76,8 +75,6 @@
   inline static MutexType sMutex;
 
   TestFileManagerStats* const mStats;
-
-  ~TestFileManager() = default;
 };
 
 using TestFileInfo = FileInfoT<TestFileManager>;
@@ -87,7 +84,7 @@
 
 TEST(DOM_IndexedDB_TestFileManager, Invalidate)
 {
-  const auto fileManager = MakeRefPtr<TestFileManager>();
+  const auto fileManager = MakeSafeRefPtr<TestFileManager>();
 
   fileManager->Invalidate();
 
@@ -103,7 +100,7 @@
   auto stats = TestFileManagerStats{};
 
   {
-    const auto fileManager = MakeRefPtr<TestFileManager>(&stats);
+    const auto fileManager = MakeSafeRefPtr<TestFileManager>(&stats);
     auto fileInfo = fileManager->CreateFileInfo();
 
     int32_t memRefCnt, dbRefCnt;
@@ -124,7 +121,7 @@
   auto stats = TestFileManagerStats{};
 
   {
-    const auto fileManager = MakeRefPtr<TestFileManager>(&stats);
+    const auto fileManager = MakeSafeRefPtr<TestFileManager>(&stats);
     fileManager->CreateDBOnlyFileInfos();
 
     for (const auto id : TestFileManager::kDBOnlyFileInfoIds) {
@@ -151,7 +148,7 @@
   auto stats = TestFileManagerStats{};
 
   {
-    const auto fileManager = MakeRefPtr<TestFileManager>(&stats);
+    const auto fileManager = MakeSafeRefPtr<TestFileManager>(&stats);
     fileManager->CreateDBOnlyFileInfos();
 
     const auto fileInfos = TransformIntoNewArray(
@@ -179,7 +176,7 @@
   auto stats = TestFileManagerStats{};
 
   {
-    const auto fileManager = MakeRefPtr<TestFileManager>(&stats);
+    const auto fileManager = MakeSafeRefPtr<TestFileManager>(&stats);
     fileManager->CreateDBOnlyFileInfos();
 
     const auto fileInfo =
@@ -201,7 +198,7 @@
 {
   auto stats = TestFileManagerStats{};
   {
-    const auto fileManager = MakeRefPtr<TestFileManager>(&stats);
+    const auto fileManager = MakeSafeRefPtr<TestFileManager>(&stats);
     fileManager->CreateDBOnlyFileInfos();
 
     auto* fileInfo = fileManager->CreateFileInfo().forget().take();
@@ -221,7 +218,7 @@
 {
   auto stats = TestFileManagerStats{};
   {
-    const auto fileManager = MakeRefPtr<TestFileManager>(&stats);
+    const auto fileManager = MakeSafeRefPtr<TestFileManager>(&stats);
 
     fileManager->Invalidate();
 
@@ -240,7 +237,7 @@
 {
   auto stats = TestFileManagerStats{};
   {
-    const auto fileManager = MakeRefPtr<TestFileManager>(&stats);
+    const auto fileManager = MakeSafeRefPtr<TestFileManager>(&stats);
 
     const auto fileInfo = fileManager->CreateFileInfo();
     Unused << fileInfo;
@@ -258,7 +255,7 @@
 {
   auto stats = TestFileManagerStats{};
   {
-    const auto fileManager = MakeRefPtr<TestFileManager>(&stats);
+    const auto fileManager = MakeSafeRefPtr<TestFileManager>(&stats);
 
     auto* fileInfo = fileManager->CreateFileInfo().forget().take();
 
